cloudferrylib/os/compute/nova_compute.py

```diff
@@ -654,3 +654,15 @@ def get_free_disk(self):
         return (hypervisor_statistics.local_gb *
                 self.config.compute.disk_allocation_ratio -
                 hypervisor_statistics.local_gb_used)
+
+    def live_migrate_vm(self, vm_id, hostname):
+        LOG.debug("migrating {vm} to {host}".format(
+            vm=vm_id,
+            host=hostname
+        ))
+        self.nova_client.servers.live_migrate(
```

Comment:
Looks like live migration fails if VM has volume attached, so we should consider detaching volume prior to migration and attaching it back once it's migrated.